roachpb: remove settings from package

RaduBerinde opened the pull request:
`pb` packages are supposed to have as little dependencies as possible, they should not depend on higher-level machinery like `settings`. This commit moves the span stats settings out of `roachpb`. This will prevent a circular dependency if we move the cluster version setting code to `settings`.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114051)
<!-- Reviewable:end -->

maryliag reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114051#-:-NijxeYI9ZEEkgzaqzFE:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian and @renatolabs)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
roachpb: remove settings from package

`pb` packages are supposed to have as little dependencies as possible,
they should not depend on higher-level machinery like `settings`.
This commit moves the span stats settings out of `roachpb`. This will
prevent a circular dependency if we move the cluster version setting
code to `settings`.

Epic: none
Release note: None

RaduBerinde commented:
TFTR!

bors r+
